app/controllers/application_controller.rb

```diff
@@ -883,14 +883,9 @@ def view_to_hash(view, fetch_data = false)
                           :icon2 => icon2}.compact
       new_row[:cells].concat(::GtlFormatter.format_cols(view, row, self))
 
-      next unless @row_button # Show a button in the last col
-
-      new_row[:cells] << {
-        :is_button => true,
-        :text      => @row_button[:label],
-        :title     => @row_button[:title],
-        :onclick   => "#{@row_button[:function]}(\"#{row['id']}\");"
-      }
+      # Append a button if @row_button is set and the button is defined in the related decorator
+      button = item.decorate.try(:gtl_button_cell, row['id'], @row_button[:label], @row_button[:title], @row_button[:function]) if @row_button
```

Comment:
```suggestion
      button = item.decorate.try(:gtl_button_cell, row['id'], @row_button.permit(:label, :function, :title)) if @row_button
```
This way you can access the params in the decorator and decrease the number of arguments.

Reply:
Even better if we move the constants from `@row_button` into the decorator and keep it here as a boolean. Let's talk about this when you're at the office.